buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -59,6 +59,11 @@ class BuildPlugin implements Plugin<Project> {
                 throw new GradleException('Gradle 2.6 or above is required to build elasticsearch')
             }
 
+            // enforce Java version
+            if (JavaVersion.current().compareTo(JavaVersion.VERSION_1_8) < 0) {
```

Comment:
or just use the isCompatibleWith. No need to use esoteric groovy language features, it makes plenty of bad choices.